tests/test_core.py

```diff
@@ -2604,12 +2604,22 @@ def test_pthread_equal(self):
     self.do_run_in_out_file_test('pthread/test_pthread_equal.cpp')
 
   @node_pthreads
-  def test_pthread_proxying(self):
+  @parameterized({
+      '': (False,),
+      'modularize': (True,),
+  })
+  def test_pthread_proxying(self, modularize):
     self.set_setting('EXIT_RUNTIME')
     self.set_setting('PROXY_TO_PTHREAD')
     self.set_setting('INITIAL_MEMORY=32mb')
+    args = []
+    if modularize:
+      self.set_setting('MODULARIZE')
+      self.set_setting('EXPORT_NAME=ModuleFactory')
+      create_file('extern-post.js', 'ModuleFactory();')
+      args = ['--extern-post-js=extern-post.js']
```

Comment:
This code will run on all threads.. is that needed/good?

Reply:
Oh, good point. That's not what I intended, and I wonder if it is the cause of the problems I'm seeing.